x-pack/solutions/security/plugins/security_solution/public/flyout/document_details/left/components/host_details.tsx

```diff
@@ -147,6 +162,41 @@ export const HostDetails: React.FC<HostDetailsProps> = ({ hostName, timestamp, s
     skip: selectedPatterns.length === 0,
   });
 
+  const filterQuery = useMemo(
```

Comment:
looking at the other files in this PR, it almost look like we could extract a lot of this logic into a reusable hook that would return the necessary information. It seems that from the`filterQuery` all the way to the `useNavigateToHostDetails` is identical to the addition in the `host_entity_overview` file.

Similar thoughts on the code added in the `user_details` and `user_entity_overview` files.

Wdyt?

Reply:
Again posting to document our discussion. Although it will look more streamlined in the code, I think the abstraction is not fundamentally necessary. The query call and misconfiguration checks can all be done in a context wrapper in the entity flyout. It makes more sense to me to do the clean up on their end, and will benefit code in both teams.